addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java

```diff
@@ -208,6 +218,16 @@ public void updateState(YamahaReceiverState state) throws IOException {
         node = getNode(basicStatus, "Input/Input_Sel_Item_Info/Src_Number");
         value = node != null ? node.getTextContent() : "0";
         state.netRadioChannel = Integer.parseInt(value);
+
+        // Get currently playing net radio station for menu based receivers
+        if (includeNetradioStation) {
```

Comment:
I'm definitely not happy with this conditional postAndGetXmlResponse. updateState() should not require any arguments, especially no boolean arguments that cause conditional network traffic (it's harder to test!). Could you instead introduce another update method like updateNetRadioMenu() or so and call that method wherever required?